protos/ledger/rwset/kvrwset/kv_rwset.proto

```diff
@@ -21,11 +21,44 @@ option java_package = "org.hyperledger.fabric.protos.ledger.rwset.kvrwset";
 
 package kvrwset;
 
-// KVRWSet encapsulates the read-write set for a chaincode that operates upon a KV or Document data model
+// KVRWSet encapsulates two read-write set for a chaincode that operates upon a KV or
+// Document data model, and the two read-write sets are 
+// (1) the primary read-write set for public data. 
+// (2) hashsed read-write set for private data (all collections). 
 message KVRWSet {
+
+    // Primary read-write set for public data
     repeated KVRead reads = 1;
     repeated RangeQueryInfo range_queries_info = 2;
     repeated KVWrite writes = 3;
+
+    // Hashed read-write set for private data
+    repeated HashedRWSet hashed_r_w = 4;
+
+}
+
+// HashedRWSet encapsulates private read-write set of a collections for a chaincode 
+// that operates upon a KV or Document data model. 
+message HashedRWSet {
+    string collection = 1;
+    repeated KVRead hashed_reads = 2;
+    string hashed_range_query_info = 3;
+    repeated KVWrite hashed_writes = 4;
+}
+
+// PrivateKVRWSet encapsulates private read-write set of all collections for a chaincode 
+// that operates upon a KV or Document data model. 
+message PrivateCollectionsKVRWSets {
+    repeated PrivateKVRWSet private_r_w_set = 1; 
+}
+
+// PrivateKVRWSet encapsulates private read-write set of a collection for a chaincode 
+// that operates upon a KV or Document data model. 
+message PrivateKVRWSet {
+    string collection = 1;
+    repeated KVRead private_reads = 2;
+    repeated RangeQueryInfo private_range_queries_info = 3;
```

Comment:
I am not sure whether RangeQuery can be supported with the collection feature. Because, the validators who do not have access to a collection, cannot validate the RWSet. So, Ideally, collection and RangeQueryInfo fields should be declared 'either/or'. But for first cut, we can keep the `collection` in and `rangequeryInfo` out for simplicity.